Marlin/src/gcode/calibrate/G34_M422.cpp

```diff
@@ -366,7 +366,7 @@ void GcodeSuite::G34() {
       // After this operation the z position needs correction
       set_axis_not_trusted(Z_AXIS);
       // Home Z after the alignment procedure
-      process_subcommands_now_P(PSTR("G28Z"));
+      process_subcommands_now_P(PSTR(TERN(Z_SAFE_HOMING, "G28", "G28Z")));
```

Comment:
Thing is, `G28` should always make sure X and Y have been homed before homing Z when `Z_SAFE_HOMING` is enabled, and the nozzle should move to the safe homing point. So the bug being worked around here is actually in `GcodeSuite::G28()`.